### DISCUSSION TITLE
Is it possible to use [slug] in api routes?

### DISCUSSION THREAD

**Discussion by gsbelarus** [vercel/next.js#60260] [App Router]
I have the following directories structure:
```
/app/github/webhook/[token]/route.ts
```
How can I get the [token] value inside `POST` request handler?

The documentation completely missed the point of using `[slugs]` inside api routes.

**Comment by icyJoseph** [comment]
Hi,

This is possible and documented here:

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers#dynamic-route-segments

Which tells you to do:

```ts
export async function GET(
  request: Request,
  { params }: { params: { slug: string } }
) {
  const slug = params.slug // 'a', 'b', or 'c'
}
```

Or with just JavaScript:

```js
export async function GET(request, { params }) {
  const slug = params.slug // 'a', 'b', or 'c'
}
```

And this table documents the routes, and outputs:



Route | Example URL | params
-- | -- | --
app/items/[slug]/route.js | /items/a | { slug: 'a' }
app/items/[slug]/route.js | /items/b | { slug: 'b' }
app/items/[slug]/route.js | /items/c | { slug: 'c' }
